File: app/src/main/java/com/example/kmmkitchentmanagement/adapter/NotificationAdapter.java
package com.example.kmmkitchentmanagement.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.kmmkitchentmanagement.Model.Notification;
import com.example.kmmkitchentmanagement.R;
import com.example.kmmkitchentmanagement.viewmodelExtends.NotificationViewModel;

import java.util.ArrayList;
import java.util.List;

public class NotificationAdapter extends BaseAdapter {
    private Context context;
    private List<Notification> notificationList;
    private NotificationViewModel notificationViewModel;

    public NotificationAdapter(Context context, List<Notification> notifications, NotificationViewModel viewModel) {
        this.context = context;
        this.notificationList = notifications != null ? notifications : new ArrayList<>();
        this.notificationViewModel = viewModel;
    }

    @Override
    public int getCount() {
        return notificationList.size();
    }

    @Override
    public Object getItem(int position) {
        return notificationList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;

        if (convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.listview_notification, parent, false);
            holder = new ViewHolder();
            holder.imageEmail = convertView.findViewById(R.id.imageEmail);
            holder.titleTextView = convertView.findViewById(R.id.titleNotificatoin);
            holder.userTextView = convertView.findViewById(R.id.userNotificatoin);
            holder.timeTextView = convertView.findViewById(R.id.timeNotification);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        // Cập nhật dữ liệu cho các view
        Notification notification = (Notification) getItem(position);
        updateView(holder, notification);

        return convertView;
    }

    private void updateView(ViewHolder holder, Notification notification) {
        if (notification.isRead()) {
            holder.imageEmail.setImageResource(R.drawable.open_message);
        } else {
            holder.imageEmail.setImageResource(R.drawable.email);
        }
        holder.titleTextView.setText(notification.getTitle());
        holder.userTextView.setText(notification.getUser());
        holder.timeTextView.setText(notification.getDateString());
    }

    public void updateNotifications(List<Notification> notifications) {
        this.notificationList.clear();
        if (notifications != null) {
            this.notificationList.addAll(notifications);
        }
        notifyDataSetChanged(); // Thông báo cho ListView để cập nhật
    }

    private static class ViewHolder {
        ImageView imageEmail;
        TextView titleTextView;
        TextView userTextView;
        TextView timeTextView;
    }
}

File: app/src/main/java/com/example/kmmkitchentmanagement/adapter/NotificationAdapter.kt
package com.example.kmmkitchentmanagement.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.example.kmmkitchentmanagement.Model.Notification
import com.example.kmmkitchentmanagement.R
import com.example.kmmkitchentmanagement.viewmodelExtends.NotificationViewModel

class NotificationAdapter(
    private val context: Context,
    private var notificationList: MutableList<Notification> = mutableListOf(),
    private val notificationViewModel: NotificationViewModel
) : BaseAdapter() {

    override fun getCount(): Int {
        return notificationList.size
    }

    override fun getItem(position: Int): Any {
        return notificationList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var holder: ViewHolder

        val view = convertView ?: LayoutInflater.from(context).inflate(R.layout.listview_notification, parent, false).apply {
            holder = ViewHolder()
            holder.imageEmail = findViewById(R.id.imageEmail)
            holder.titleTextView = findViewById(R.id.titleNotificatoin)
            holder.userTextView = findViewById(R.id.userNotificatoin)
            holder.timeTextView = findViewById(R.id.timeNotification)
            tag = holder
        }

        holder = view.tag as ViewHolder

        val notification = getItem(position) as Notification
        updateView(holder, notification)

        return view
    }

    private fun updateView(holder: ViewHolder, notification: Notification) {
        if (notification.isRead) {
            holder.imageEmail.setImageResource(R.drawable.open_message)
        } else {
            holder.imageEmail.setImageResource(R.drawable.email)
        }
        holder.titleTextView.text = notification.title
        holder.userTextView.text = notification.user
        holder.timeTextView.text = notification.dateString
    }

    fun updateNotifications(notifications: List<Notification>?) {
        notificationList.clear()
        notifications?.let { notificationList.addAll(it) }
        notifyDataSetChanged()
    }

    private class ViewHolder {
        lateinit var imageEmail: ImageView
        lateinit var titleTextView: TextView
        lateinit var userTextView: TextView
        lateinit var timeTextView: TextView
    }
}
